test/types/index.ts

```diff
@@ -30,12 +33,28 @@ import { createReadStream, readFile } from 'fs'
   });
   //logger true
   const logAllServer = fastify({ logger: true });
+  logAllServer.addHook('onRequest', (req, res, next) => {
+    console.log('can access req', req.headers);
+    next();
+  });
 
   // custom types
   interface CustomIncomingMessage extends http.IncomingMessage {
     getDeviceType: () => string;
   }
   const customServer: fastify.FastifyInstance<http.Server, CustomIncomingMessage, http.ServerResponse> = fastify();
+  customServer.use((req, res, next) => {
+    console.log('can access props from CustomIncomingMessage', req.getDeviceType());
+  })
+
+  interface CustomHttp2IncomingMessage extends http2.Http2ServerRequest {
+    getDeviceType: () => string;
+  }
+
+  const customHttp2Server: fastify.FastifyInstance<http2.Http2Server, CustomHttp2IncomingMessage, http2.Http2ServerResponse> = fastify();
+  customHttp2Server.use((req, res, next) => {
+    console.log('can access props from CustomIncomingMessage', req.getDeviceType());
+  });
```

Comment:
why are you adding this test?

Reply:
This is testing that you can create set the generic types to be an `http2` server.
The `use` and `log` are ensuring that the generic request type is passed along when executing `.use((req, res, next) => {})`

Reply:
Thanks!